### PR TITLE
chore: Update tests to avoid running `setup test`

### PR DESCRIPTION
**Pull request by fsbraun**
## Summary by Sourcery

Update CI configuration to use the latest Python and Django versions, adjust test execution to use 'tests/settings.py', and ensure compatibility by modifying the Python path in the test settings.

CI:
- Update GitHub Actions workflow to use the latest Ubuntu version and expand the matrix to include Python 3.11, 3.12, and 3.13, and Django 4.2, 5.0, and 5.1.

Tests:
- Modify test execution to run coverage using 'tests/settings.py' instead of 'setup.py test'.
- Update 'tox.ini' to reflect changes in Django version dependencies and test execution command.

Chores:
- Add current directory to Python path in 'tests/settings.py' to ensure correct script execution.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -11,14 +11,14 @@ skip_missing_interpreters=True
 deps =
     -r{toxinidir}/tests/requirements/base.txt
     dj32: Django>=3.2,<3.3
-    dj40: Django>=4.0,<4.1
-    dj41: Django>=4.1,<4.2
-    dj42: Django>=4.2,<5.0
+    dj42: Django>=4.2,<4.3
+    dj50: Django>=5.0,<4.1
```
Comment (fsbraun) [inline tox.ini:15]:
```suggestion
    dj50: Django>=5.0,<5.1
```

File: tests/settings.py [hunk 2]
```diff
@@ -59,4 +59,7 @@ def run():
 
 
 if __name__ == '__main__':
+    import sys
+     # Add current directory to python path - works if this script is called python tests/settings,.py
```
Comment (fsbraun) [inline tests/settings.py:63]:
```suggestion
    # Add current directory to python path - works if this script is called python tests/settings,.py
```

File: tests/settings.py [hunk 3]
```diff
@@ -59,4 +59,7 @@ def run():
 
 
 if __name__ == '__main__':
+    import sys
```
Comment (fsbraun) [inline tests/settings.py:62]:
```suggestion
```

**Comment by fsbraun** [comment]
@sourcery-ai review

**Review by vinitkumar — APPROVED** [review]

**Pull request merged**